crates/red_knot_python_semantic/resources/mdtest/terminal_statements.md

```diff
@@ -0,0 +1,638 @@
+    else:
+        reveal_type(x)  # revealed: Literal["else1", "else2"]
+    finally:
+        # This includes "before" because we assume that an exception might have occurred before the
+        # `if` statement.
+        reveal_type(x)  # revealed: Literal["before", "else1", "raise", "else2"]
+    reveal_type(x)  # revealed: Literal["before", "else1", "raise", "else2"]
+
+def raise_in_nested_else_branch(cond1: bool, cond2: bool):
+    x = "before"
+    try:
+        if cond1:
+            x = "else1"
+            reveal_type(x)  # revealed: Literal["else1"]
+        else:
+            if cond2:
+                x = "else2"
+                reveal_type(x)  # revealed: Literal["else2"]
+            else:
+                x = "raise"
+                reveal_type(x)  # revealed: Literal["raise"]
+                raise ValueError
+            reveal_type(x)  # revealed: Literal["else2"]
+        reveal_type(x)  # revealed: Literal["else1", "else2"]
+    except ValueError:
+        # TODO: Literal["raise"]
+        reveal_type(x)  # revealed: Literal["before", "else1", "else2", "raise"]
+    except:
+        # TODO: Literal["raise"] or Never
+        reveal_type(x)  # revealed: Literal["before", "else1", "else2", "raise"]
+    else:
+        reveal_type(x)  # revealed: Literal["else1", "else2"]
+    finally:
+        # This includes "before" because we assume that an exception might have occurred before the
+        # `if` statement.
+        reveal_type(x)  # revealed: Literal["before", "else1", "else2", "raise"]
+    reveal_type(x)  # revealed: Literal["before", "else1", "else2", "raise"]
+
+def raise_in_both_nested_branches(cond1: bool, cond2: bool):
+    x = "before"
+    try:
+        if cond1:
+            x = "else"
+            reveal_type(x)  # revealed: Literal["else"]
+        else:
+            if cond2:
+                x = "raise1"
+                reveal_type(x)  # revealed: Literal["raise1"]
+                raise ValueError
+            else:
+                x = "raise2"
+                reveal_type(x)  # revealed: Literal["raise2"]
+                raise ValueError
+        reveal_type(x)  # revealed: Literal["else"]
+    except ValueError:
+        # TODO: Literal["raise1", "raise2"]
+        reveal_type(x)  # revealed: Literal["before", "else", "raise1", "raise2"]
+    except:
+        # TODO: Literal["raise1", "raise2"] or Never
+        reveal_type(x)  # revealed: Literal["before", "else", "raise1", "raise2"]
+    else:
+        reveal_type(x)  # revealed: Literal["else"]
+    finally:
+        # This includes "before" because we assume that an exception might have occurred before the
+        # `if` statement.
+        reveal_type(x)  # revealed: Literal["before", "else", "raise1", "raise2"]
+    reveal_type(x)  # revealed: Literal["before", "else", "raise1", "raise2"]
+```
+
+## Terminal in a `finally` block
+
+Control-flow through finally isn't working right yet:
```

Comment:
This also seems overly broad; we could be more specific to save future-us from possible confusion about what we meant
```suggestion
TODO: we don't yet model that a `break` or `continue` in a `try` block will jump to a `finally` clause before it
jumps to end/start of the loop.
```